### DISCUSSION TITLE
Listbox should have events for opened, closing

### DISCUSSION THREAD

**Discussion by DarkGhostHunter** [tailwindlabs/headlessui#2170] [Ideas]
For example, we can use this to integrate popper.js.

```vue
<template>
   <Listbox ... @onOpened="integratePopper" @onClosed="removePopper">
       <ListboxOptions ref="listboxOptionsEl">
            ...
       </ListboxOptions>
   </Listbox>
</template>

<script setup>
import { createPopper } from '@popperjs/core'

const listboxOptionsEl = ref(null)

function integratePopper(event) {
    createPopper(event.target, listboxOptionsEl)
}
</script>
```

- `onOpen` = Before the ListboxOptions is mounted/shown.
- `onOpened` = After the ListboxOptions is mounted/shown.
- `onClose` = Before the ListboxOptions is unmounted/hidden.
- `onClosed` = After the ListboxOptions is unmounted/hidden.

**Comment by olivierpascal** [comment]
Yes, please add open/close events for components
